feat: allow users to pass true to videoCompression config and only a… (#26810)
cypress-io/cypress: cli/CHANGELOG.md

<!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->

## 12.14.1

_Released 06/20/2023 (PENDING)_

**Bugfixes:**

> - The `videoCompression` option now only accepts a boolean or a valid CRF options between 0-51 (inclusively). In addition, `true` is now a valid option, which will default to 32 CRF. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).

Author: ```suggestion
- The [`videoCompression`](https://docs.cypress.io/guides/references/configuration#Videos) configuration option now accepts both a boolean or a Constant Rate Factor (CRF) number between `0` and `51`. The `videoCompression` default value is still `32` CRF and when `videoCompression` is set to `true` the default of `32` CRF will be used. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).
```

> - The [`videoCompression`](https://docs.cypress.io/guides/references/configuration#Videos) configuration option now accepts both a boolean or a Constant Rate Factor (CRF) number between `0` and `51`. The `videoCompression` default value is still `32` CRF and when `videoCompression` is set to `true` the default of `32` CRF will be used. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).

Reviewer: ```suggestion
- The [`videoCompression`](https://docs.cypress.io/guides/references/configuration#Videos) configuration option now accepts both a boolean or a Constant Rate Factor (CRF) number between `1` and `51`. The `videoCompression` default value is still `32` CRF and when `videoCompression` is set to `true` the default of `32` CRF will be used. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).
```

Author: 0 is still a valid number for CRF, we just don't compress the video since a CRF of 0 is essentially a non compressed video

Reviewer: I think its better to be explicit that this disabled compression even if it is a valid CRF value. If I didn't have context, I'd expect 0 to compress since false is called out to prevent compression.

Author: I think I got this added in 53a07c986d

> - The [`videoCompression`](https://docs.cypress.io/guides/references/configuration#Videos) configuration option now accepts both a boolean or a Constant Rate Factor (CRF) number between `0` and `51`. The `videoCompression` default value is still `32` CRF and when `videoCompression` is set to `true` the default of `32` CRF will be used. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).

Author: @emily how is something like this?
```suggestion
- The [`videoCompression`](https://docs.cypress.io/guides/references/configuration#Videos) configuration option now accepts both a boolean or a Constant Rate Factor (CRF) number between `0` and `51`. The `videoCompression` default value is still `32` CRF and when `videoCompression` is set to `true` the default of `32` CRF will be used. If a CRF of `0` is provided, compression is skipped as the video is already considered lossless and equivalent to a CRF of `0`. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).
```

- Fixed an issue to now correctly detect Angular 16 dependencies during Component Testing onboarding. Addresses [#26852](https://github.com/cypress-io/cypress/issues/26852)
- Ensures Git-related messages on the [Runs page](https://docs.cypress.io/guides/core-concepts/cypress-app#Runs) remain dismissed. Addresses [#26808](https://github.com/cypress-io/cypress/issues/26808)

> - The `videoCompression` option now only accepts a boolean or a valid CRF options between 1-51 (inclusively). In addition, `true` is now a valid option, which will default to 32 CRF. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).

Author: ```suggestion
- The `videoCompression` option now only accepts a boolean or a valid CRF options between 0-51 (inclusively). In addition, `true` is now a valid option, which will default to 32 CRF. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).
```

> - The `videoCompression` option now only accepts a boolean or a valid CRF options between 0-51 (inclusively). In addition, `true` is now a valid option, which will default to 32 CRF. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).

Reviewer: ```suggestion
- The [`videoCompression`](https://docs.cypress.io/guides/references/configuration#Videos) configuration option now accepts both a boolean or a Constant Rate Factor (CRF) number between `0` and `51`. The `videoCompression` default value is still `32` CRF and when `videoCompression` is set to `true` the default of `32` CRF will be used. Addresses [#26658](https://github.com/cypress-io/cypress/issues/26658).
```

Author: addressed in 53a07c986d
